### PR TITLE
react-select: minor style cleanup in select stories

### PR DESCRIPTION
**Pull request by smhigley**
A few minor story-only style changes to improve the spacing in select stories.

### CODE REVIEW

File: packages/react-components/react-select/src/stories/SelectControlled.stories.tsx [hunk 1]
```diff
@@ -1,21 +1,39 @@
...
+import { makeStyles } from '@griffel/react';
+
+const useStyles = makeStyles({
+  root: {
+    maxWidth: '400px',
+
+    // Stack the label above the field with a 2px gap
+    '> label': {
+      display: 'block',
+      marginBottom: '2px',
```
Comment (spmonahan) [inline packages/react-components/react-select/src/stories/SelectControlled.stories.tsx:13]: I'd recommend you use a vertical spacing token so the story better demonstrates best practices.

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `181.156 kB`<br />`50.604 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `34.078 kB`<br />`11.098 kB` |
| <samp>react-select</samp> <br /> <abbr title='bundle-size/Select.fixture.js'>Select</abbr>                                                                                    |   `17.138 kB`<br />`6.367 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/88618d9aa570789cca97a3b22921232fc8252aed'>88618d9aa570789cca97a3b22921232fc8252aed</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23210/perf-test-react-components/Avatar-mount_base.html">1079</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23210/perf-test-react-components/Avatar-mount.html">1102</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23210/perf-test-react-components/Button-mount_base.html">633</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23210/perf-test-react-components/Button-mount.html">690</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23210/perf-test-react-components/FluentProvider-mount_base.html">2242</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23210/perf-test-react-components/FluentProvider-mount.html">2258</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23210/perf-test-react-components/FluentProviderWithTheme-mount_base.html">330</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23210/perf-test-react-components/FluentProviderWithTheme-mount.html">315</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23210/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">292</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23210/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">284</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23210/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">390</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23210/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">369</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23210/perf-test-react-components/MakeStyles-mount_base.html">1880</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23210/perf-test-react-components/MakeStyles-mount.html">1893</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Review by spmonahan — APPROVED** [review]

**Review by khmakoto — APPROVED** [review]

File: packages/react-components/react-select/src/stories/SelectSize.stories.tsx [hunk 2]
```diff
@@ -1,12 +1,29 @@
...
 import { Select } from '../index';
+import { tokens } from '@fluentui/react-theme';
 import { useId } from '@fluentui/react-utilities';
+import { makeStyles } from '@griffel/react';
+
+const useStyles = makeStyles({
+  root: {
+    maxWidth: '400px',
+
+    '> * + label': { marginTop: tokens.spacingVerticalMNudge },
```
Comment (layershifter) [inline packages/react-components/react-select/src/stories/SelectSize.stories.tsx:11]: Can we use `display: grid` and `gap` instead?

Comment (spmonahan) [reply packages/react-components/react-select/src/stories/SelectSize.stories.tsx:11]: I've actually changed this for SpinButton based on similar comments.

https://github.com/microsoft/fluentui/blob/master/packages/react-components/react-spinbutton/src/stories/SpinButtonSize.stories.tsx#L8

Comment (layershifter) [reply packages/react-components/react-select/src/stories/SelectSize.stories.tsx:11]: > Is there a benefit to grid over margin in this case?

The point of stories it to how consumers how to use components and let them copy-paste code. IMO `'> * + label'` as a selector is not something that I would recommend doing.

I would recommend to do:

```ts
const useStyles = makeStyles({
  root: { maxWidth: '400px' },
  label: { marginTop: tokens.spacingVerticalMNudge },
  // etc.
})
```

And avoid nested selectors if they are not needed.

Comment (smhigley) [reply packages/react-components/react-select/src/stories/SelectSize.stories.tsx:11]: In a layout use case like this, modified versions of the lobotomized owl selector are actually exactly what I would recommend consumers use, at least if grid/flexbox weren't a good fit 😄. If I were writing styles that were specific to the labels themselves, I'd definitely add `label` to `makeStyles()`. This seems like a case where the parent should be able to handle its own layout for children without the children needing to modify themselves to participate.

Maybe a good approach to demo to consumers might be to import `selectClassNames.root` and use it here in place of the `*` selector, partly for marginally better selector performance, and partly to show consumers how to use it.

@spmonahan I have a very slight personal preference towards using `+` over using `:first-child { marginTop: 0 }`, but I'm also happy to switch to the latter. Was there any particular piece of feedback towards the latter?

Comment (layershifter) [reply packages/react-components/react-select/src/stories/SelectSize.stories.tsx:11]: Please note following:

- Complex selectors harder to override due specificity
- CSS that comes from parent is harder to override (find parent first 🕹)
- The goal of Atomic CSS is to have less unique rules across an app: `.a { color: red }` & `.b * { color: red }` are different rules => more CSS output => more bundle size. It's one of the reasons to keep our styles flat

**Review by GeoffCoxMSFT — APPROVED** [review]

**Pull request closed without merging**